src/plone/restapi/serializer/__init__.py

```diff
@@ -0,0 +1,101 @@
+
+from plone.app.contenttypes.interfaces import ICollection
+from plone.restapi.interfaces import ISerializeToJson
+from plone.restapi.interfaces import IFieldSerializer
+from plone.restapi.serializer.converters import json_compatible
+from plone.dexterity.utils import iterSchemata
+
+from zope.site.hooks import getSite
+from zope.interface import implementer
+from zope.component import adapter
+from zope.component import getMultiAdapter
+from zope.component import queryMultiAdapter
+from zope.schema import getFields
+
+
+def lookup_field_serializer(context, field):
+    serializer = queryMultiAdapter((field, context, context.REQUEST),
+                                   IFieldSerializer,
+                                   name=field.interface.__identifier__)
+    if serializer:
+        return serializer
+
+    serializer = queryMultiAdapter((field, context, context.REQUEST),
+                                   IFieldSerializer,
+                                   name=field.__name__)
+    if serializer:
+        return serializer
+
+    return getMultiAdapter((field, context, context.REQUEST),
+                           IFieldSerializer)
+
+
+@implementer(ISerializeToJson)
+@adapter(IPloneSiteRoot)
+def SerializeSiteRootToJson(context):
+    result = {
+        '@context': 'http://www.w3.org/ns/hydra/context.jsonld',
+        '@id': context.absolute_url(),
+        '@type': 'SiteRoot',
+        'parent': {},
+    }
+    result['member'] = [
+        {
+            '@id': member.absolute_url(),
+            'title': member.title,
+            'description': member.description
+        }
+        for member in context.objectValues()
+        if IContentish.providedBy(member)
+    ]
+    return result
+
+
+@implementer(ISerializeToJson)
+@adapter(IContentish)
+def SerializeToJson(context):
+    result = {
+        '@context': 'http://www.w3.org/ns/hydra/context.jsonld',
+        '@id': context.absolute_url(),
+        '@type': context.portal_type,
+        'parent': {
+            '@id': aq_parent(aq_inner(context)).absolute_url(),
+            'title': aq_parent(aq_inner(context)).title,
+            'description': aq_parent(aq_inner(context)).description
+        }
+    }
+    if IFolderish.providedBy(context):
+        result['member'] = [
+            {
+                '@id': member.absolute_url(),
+                'title': member.title,
+                'description': member.description
+            }
+            for member in context.objectValues()
+        ]
+    if ICollection.providedBy(context):
+        portal = getSite()
+        result['member'] = [
+            {
+                '@id': '{0}/{1}'.format(
+                    portal.absolute_url(),
+                    '/'.join(member.getPhysicalPath())
+                ),
+                'title': member.title,
+                'description': member.description
+            }
+            for member in context.results()
+        ]
+
+    for schemata in iterSchemata(context):
```

Comment:
`schemata` -> `schema` (singluar). (yep, it's been copy and pasted all over the place like that, for that reason we could also just leave it be)